quill-sql/src/test/scala/io/getquill/context/sql/idiom/PostgresDialectSpec.scala

```diff
@@ -24,5 +24,13 @@ class PostgresDialectSpec extends Spec {
       }
       context.run(q).string mustEqual "SELECT t.s::integer FROM TestEntity t"
     }
+    "upsert" in {
+      val e = TestEntity("", 1, 1L, Some(1))
+      val q = quote {
+        query[TestEntity].upsert(lift(e)).conflict(_.i).conflictUpdate(_.i -> lift(2), _.l -> lift(1L), _.s -> lift("Test String"))
```

Comment:
it doesn't seem that `conflict` can be used without `conflictUpdate`, I think we could simplify things and avoid misuse:

```scala
query[TestEntity].insert(lift(e)).onConflict(_.i)(_.i -> lift(2), _.l -> lift(1L), _.s -> lift("Test String"))
```

The AST could also be simplified to a single new type. Wdyt?